### PR TITLE
docs: removing Template API

### PR DESCRIPTION
**Pull request by vursen**
- Adds a section to the Upgrading Guide explaining that the Template API (`<template>`) is no longer supported and renderer functions or the template renderer package should be used instead.
- Removes any mentions of Template API support
- Converts templates to renderer functions in the Fusion examples

Closes https://github.com/vaadin/web-components/issues/2100
Closes https://github.com/vaadin/web-components/issues/2113

### CODE REVIEW

File: articles/guide/upgrading.asciidoc [hunk 1]
```diff
@@ -46,6 +46,43 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
 
 The following instructions apply when upgrading from a version before Vaadin 21.
 
+=== The Web Components No Longer Support <template>
```
Comment (vursen) [inline articles/guide/upgrading.asciidoc:49]: The Vaadin Web Components?

File: articles/fusion/application/basics.asciidoc [hunk 2]
```diff
@@ -161,11 +163,16 @@ export class MyView extends LitElement {
...
-      <vaadin-notification id="notification" duration="2000">
-        <template>
-          Hello, World!
-        </template>
-      </vaadin-notification>
+      <vaadin-notification
+        id="notification"
+        duration="2000"
+        .renderer="${guard([], () => (root: HTMLElement) => {
+          render(
```
Comment (tomivirkki) [inline articles/fusion/application/basics.asciidoc:170]: Import to `render` is missing

File: articles/guide/upgrading.asciidoc [hunk 3]
```diff
@@ -46,6 +46,43 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
...
+
+As a part of migration to LitElement, the Web Components no longer support `<template>` to render a content.
```
Comment (tomivirkki) [inline articles/guide/upgrading.asciidoc:51]: Maybe LitElement shouldn't be mentioned here yet as the components are still Polymer based.

File: articles/fusion/application/basics.asciidoc (hunk 2 above)
Comment (vursen) [reply articles/fusion/application/basics.asciidoc:170]: Thanks! Fixed.

File: articles/guide/upgrading.asciidoc (hunk 3 above)
Comment (vursen) [reply articles/guide/upgrading.asciidoc]: Removed the mention.

File: articles/guide/upgrading.asciidoc (hunk 1 above)
Comment (vursen) [reply articles/guide/upgrading.asciidoc]: Changed the sentence to use "The Vaadin Web Components" instead.

File: articles/guide/upgrading.asciidoc [hunk 4]
```diff
@@ -46,6 +46,54 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
...
+```
+
+Import `vaadin-template-renderer` before any other components:
+
+[source,typescript]
+----
+import '@vaadin/vaadin-template-renderer';
+----
+
+Now, you can use templates as before:
```
Comment (vursen) [inline articles/guide/upgrading.asciidoc:71]: Is it necessary?

Comment (tomivirkki) [reply articles/guide/upgrading.asciidoc:71]: Could be removed.

File: articles/guide/upgrading.asciidoc [hunk 5]
```diff
@@ -46,6 +46,54 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
...
+Install `vaadin-template-renderer`:
+```
+npm i @vaadin/vaadin-template-renderer --save
+```
+
+Import `vaadin-template-renderer` before any other components:
+
+[source,typescript]
+----
+import '@vaadin/vaadin-template-renderer';
```
Comment (tomivirkki) [inline articles/guide/upgrading.asciidoc:68]: Could add the import for vaadin-combo-box here to showcase that it should be added after the template-renderer import?

Comment (tomivirkki) [reply articles/guide/upgrading.asciidoc:68]: Ah, I see you removed the, nevermind then

File: articles/guide/upgrading.asciidoc (hunk 4 above)
Comment (vursen) [reply articles/guide/upgrading.asciidoc]: Removed.

File: articles/guide/upgrading.asciidoc (hunk 5 above)
Comment (vursen) [reply articles/guide/upgrading.asciidoc:68]: Added, please check it.

Comment (vursen) [reply articles/guide/upgrading.asciidoc:68]: Removed :)

File: articles/guide/upgrading.asciidoc [hunk 6]
```diff
@@ -46,6 +46,43 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
...
+----
+
+==== Deprecation Warning
+
+By default, `vaadin-template-renderer` shows a deprecation warning once a component using `<template>`.
```
Comment (tomivirkki) [inline articles/guide/upgrading.asciidoc:73]:
```suggestion
By default, `vaadin-template-renderer` shows a deprecation warning when `<template>` is used with a component.
```

Comment (vursen) [reply articles/guide/upgrading.asciidoc]: Thanks! Fixed.

**Review by tomivirkki — APPROVED** [review]

File: articles/guide/upgrading.asciidoc [hunk 7]
```diff
@@ -46,6 +46,43 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
 
 The following instructions apply when upgrading from a version before Vaadin 21.
 
+=== The Vaadin Web Components No Longer Support <template>
+
+The Vaadin Web Components no longer support `<template>` to render content.
+
+Please, use renderer functions instead.
+
+Alternatively, you can use the `@vaadin/vaadin-template-renderer` package which is created to maintain backward compatibility.
```
Comment (magi42) [inline articles/guide/upgrading.asciidoc:55]:
```suggestion
The Vaadin Web Components no longer support `<template>` to render content.
Please, use renderer functions instead.
Alternatively, you can use the `@vaadin/vaadin-template-renderer` package which is created to maintain backward compatibility.
```

File: articles/guide/upgrading.asciidoc [hunk 8]
```diff
@@ -46,6 +46,43 @@ include::{root}/pom.xml[tag=spring-version, indent=0]
...
+
+==== Installation
+
+Install `vaadin-template-renderer`:
```
Comment (magi42) [inline articles/guide/upgrading.asciidoc:59]:
```suggestion
Install `vaadin-template-renderer` as follows:
```

**Review by magi42 — APPROVED** [review]

**Pull request merged**